.github/workflows/maven_test.yml

```diff
@@ -159,20 +159,22 @@ jobs:
         with:
           distribution: zulu
           java-version: ${{ matrix.java }}
-      - name: Install Python 3.9
+      - name: Install Python 3.11
         uses: actions/setup-python@v5
         # We should install one Python that is higher than 3+ for SQL and Yarn because:
         # - SQL component also has Python related tests, for example, IntegratedUDFTestUtils.
         # - Yarn has a Python specific test too, for example, YarnClusterSuite.
-        if: contains(matrix.modules, 'resource-managers#yarn') || (contains(matrix.modules, 'sql#core')) || contains(matrix.modules, 'connect')
+        # macos (14) already has its Python installed, see also SPARK-47096 and
+        # https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md
+        if: contains(inputs.os, 'ubuntu') && (contains(matrix.modules, 'resource-managers#yarn') || (contains(matrix.modules, 'sql#core')) || contains(matrix.modules, 'connect'))
```

Comment:
?

This looks like upgrade `Ubuntu` OS instead of `MacOS 14` in order to match with all Maven CIs. Shall we revise the current PR title?
> [SPARK-47096][INFRA] Upgrade Python version in Maven build in macos-14 build

Reply:
Oops yes